pkg/kbuffer/dedupe.go

```diff
@@ -25,36 +25,36 @@ import (
 var shuttingDownError = ActivationResult{
 	Endpoint: Endpoint{},
 	Status:   http.StatusInternalServerError,
-	Error:    fmt.Errorf("activator shutting down"),
+	Error:    fmt.Errorf("kbuffer shutting down"),
 }
 
-var _ Activator = (*dedupingActivator)(nil)
+var _ KBuffer = (*dedupingKBuffer)(nil)
 
-type dedupingActivator struct {
+type dedupingKBuffer struct {
 	mux             sync.Mutex
 	pendingRequests map[revisionID][]chan ActivationResult
-	activator       Activator
+	activator       KBuffer
 	shutdown        bool
 }
 
-// NewDedupingActivator creates an Activator that deduplicates
+// NewDedupingKBuffer creates a KBuffer that deduplicates
 // activations requests for the same revision id and namespace.
-func NewDedupingActivator(a Activator) Activator {
-	return &dedupingActivator{
+func NewDedupingKBuffer(a KBuffer) KBuffer {
```

Comment:
Same as below, this feels rather weird. We should maybe rename all of these types to something more neutral?